### ISSUE TITLE
Meta github check

### ISSUE THREAD

**Issue by zimbatm** [garnix-io/issues#20]
The list of github "checks" that garnix will set on a commit is variable. It depends on the shape of the flake output.

## Problem statement

In the github branch protection setting, I can tell github to only merge a PR if certain checks are passing. And I can add the checks that Garnix set during the last run in there.

The only issue is that the list of Garnix checks evolves over time. As I add more packages, I will get more checks. And there is no way to tell GitHub to ask for "all the checks". So I have to go back in there and update the list as it changes.

## Proposed solution

Have Garnix also publish a "all the garnix checks passed" check. Then I only have to add that one to the GitHub branch protection setting.

**Comment by Mic92** [comment]
Yeah. That is really needed. For example I have to manually update a long list of packages when using things like mergify: https://github.com/Mic92/sops-nix/blob/master/.mergify.yml

**Comment by Mic92** [comment]
An `all the garnix checks passed` would make my life a lot easier.

**Comment by jkarni** [comment]
I imagine changing the behavior of the `Build starting` build is also an option? I'm inclined to think that's slightly nicer UX, since then you have a single build that summarizes all others ("cloning" -> "evaluating builds" -> "building" -> "done")

**Comment by Mic92** [comment]
Hercules actually solves this nicely: https://github.com/nix-community/infra/pull/227

**Comment by Mic92** [comment]
If you have a lot of packages the list might become too long anyway.

**Comment by jkarni** [comment]
> Hercules actually solves this nicely: https://github.com/nix-community/infra/pull/227
I'm a bit unclear on what's going on there - can you explain?

**Comment by Mic92** [comment]
It's a bit confusing because mergifio also added its status updates there.

But basically hercules ci add these three status updates that are updated in this order:

1. This is when all builds finished:
- @hercules-ci ci/hercules/derivations — Success!  [Details](https://hercules-ci.com/github/nix-community/infra/jobs/292)

2. This is when evaluation is finished:
- @hercules-ci ci/hercules/evaluation — 12 attributes [Details](https://hercules-ci.com/github/nix-community/infra/jobs/292)

4. This is the impure bit of hercules ci that allows deploying (garnix.io does not have that)
- @hercules-ci ci/hercules/effects — Job has no effects [Details](https://hercules-ci.com/github/nix-community/infra/jobs/292)

**Comment by robx** [comment]
Screenshot for the previous comment (was a bit unclear where this shows):

<img width="552" alt="Screenshot 2022-07-15 at 09 07 21" src="https://user-images.githubusercontent.com/132113/179170792-d0b14b8e-57f2-4445-9bd0-3cb1d58f5594.png">

**Comment by zimbatm** [comment]
Single status is probably good enough and would keep things simple. I can't think of a case where an evaluation check is interesting to wait on.

**Comment by sellout** [comment]
I would like something like this, too (not least because the GH interface for changing required statuses is painful.

It would be cool if the checks included in the meta-check could be configured via garnix.yaml, somehow indicating which builds are required, like
```yaml
builds:
  include:
    - '*.aarch64-linux':
        required: true
    - '*.aarch64-linux.*'
```
But this could perhaps lead to issues where a malicious PR could disable required checks in the commit that are missed during review (but I think the meta-check already has this issue, which is why, IIUC, the GH required checks aren’t configurable in the code).

**Comment by sellout** [comment]
garnix now has an “All Garnix checks” check that I think satisfies this issue.

**Comment by nhooey** [comment]
Seems related to #119.
